src/server/system_services/pool_server.js

```diff
@@ -88,12 +88,27 @@ function set_pool_controller_factory(pool_controller_factory) {
 
 }
 
-function new_pool_defaults(name, system_id, resource_type, pool_node_type) {
+// The function checks whether the pool/resource has an owner
+// and only allows deletion in case that the owner is also the requester of the deletion
+function check_deletion_ownership(req, resource_owner_id) {
+    if (config.RESTRICT_RESOURCE_DELETION) {
+        const requester_is_sys_owner = String(req.account._id) === String(req.system.owner._id);
+        if (!requester_is_sys_owner && String(resource_owner_id) !== String(req.account._id)) {
+            dbg.error('check_deletion_ownership: requester (', req.account._id, ') is not the owner (', resource_owner_id, ') of the resource');
+            throw new RpcError('UNAUTHORIZED', 'The pool or resource can be deleted only by its owner or the system administrator');
+        }
```

Comment:
no reason for a return value here. in the tests you can either call the function that will throw if failed, or use `assert.doesNotThrow()` to be more explicit